src/main/java/uk/ac/ebi/eva/pipeline/jobs/steps/processors/ExcludeStructuralVariantsProcessor.java

```diff
@@ -0,0 +1,99 @@
+/*
+ *
+ * Copyright 2024 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.eva.pipeline.jobs.steps.processors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ItemProcessor;
+import uk.ac.ebi.eva.commons.models.data.Variant;
+
+import java.text.MessageFormat;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Direct implementation of VCF specification grammars from
+ * <a href="https://github.com/EBIvariation/vcf-validator/blob/master/src/vcf/vcf.ragel">here</a> and
+ * <a href="https://github.com/EBIvariation/vcf-validator/blob/master/src/vcf/vcf_v43.ragel">here</a>.
+ */
+public class ExcludeStructuralVariantsProcessor implements ItemProcessor<Variant, Variant> {
+
+    private static final Logger logger = LoggerFactory.getLogger(ExcludeStructuralVariantsProcessor.class);
+
+    private static String meta_contig_char = "(\\p{Alnum}|[\\p{Punct}&&[^:<>\\[\\]*=,]])";
+
+    private static String chromBasicRegEx = MessageFormat.format("([{0}&&[^#]][{0}]*)", meta_contig_char);
+
+    private static String chromContigRegEx = String.format("(<%s>)", chromBasicRegEx);
+
+    private static String chromosomeRegEx = String.format("(%s|%s)", chromBasicRegEx, chromContigRegEx);
+
+    private static String positionRegEx = "([\\p{Digit}]+)";
+
+    private static String basesRegEx = "([ACTGNactgn]+)";
+
+    private static String altIDRegEx_positive_match = "([\\p{Alnum}|[\\p{Punct}&&[^,<>]]]+)";
+
+    private static String altIDRegEx_negative_match = "([\\p{Punct}]+)";
+
+    private static String altIDRegEx = String.format("((?!%s)%s)", altIDRegEx_negative_match,
+            altIDRegEx_positive_match);
+
+    private static String stdPrefixRegEx = MessageFormat.format(
+            "<DEL>|<INS>|<DUP>|<INV>|<CNV>|<DUP:TANDEM>|<DEL:ME:{0}>|<INS:ME:{0}>", "(\\p{Alnum})+");
```

Comment:
Also, just noticed the example `15_100509723_G_<CNV:0>` in the other ticket [EVA3690](https://www.ebi.ac.uk/panda/jira/browse/EVA-3690), does this get covered by one of the regex?

Reply:
Yes, it does. updated the test as well to include it

Reply:
thanks!